core/imodel-bridge/src/BridgeRunner.ts

```diff
@@ -479,9 +477,13 @@ class BriefcaseDbBuilder extends IModelDbBuilder {
       throw new Error("Must initialize ContextId before using");
     if (this._serverArgs.iModelId === undefined)
       throw new Error("Must initialize IModelId before using");
-
-    // First, download the briefcase
-    const props = await BriefcaseManager.downloadBriefcase(this._requestContext, { contextId: this._serverArgs.contextId, iModelId: this._serverArgs.iModelId });
+    let props: LocalBriefcaseProps;
+    if (this._bridgeArgs.argsJson && this._bridgeArgs.argsJson.BriefcaseId) {
+      props = await BriefcaseManager.downloadBriefcase(this._requestContext, {briefcaseId: this._bridgeArgs.argsJson.BriefcaseId, contextId: this._serverArgs.contextId, iModelId: this._serverArgs.iModelId});
+    } else {
+      props = await BriefcaseManager.downloadBriefcase(this._requestContext, {contextId: this._serverArgs.contextId, iModelId: this._serverArgs.iModelId});
+      this._bridgeArgs.argsJson= {BriefcaseId: props.briefcaseId};

```

Comment:
Won't this overwrite the additional parameters passed in.

Reply:
Passing it back in there is how we're returning the id so we can save the value to the settings service in the wrapper

Reply:
Can't you instead do 'this._bridgeArgs.argsJson.BriefcaseId = props.briefcaseId' which preserves anything else previously set, and adds the id?

Reply:
I had that but when i tested it I got an error because argsJson was undefined when I didn't pass anything in, I could null check it and only do it this way when its null and otherwise do it that way if we think thats better